### ISSUE TITLE
Certain `model` names cause clashes in generated types

### ISSUE THREAD

**Issue by austincrim** [prisma/prisma#12469]
### Bug description

**existing issues**
- #12332 
- #10490
- #9773
- #9669
- #9568, #7518

The following are all exported types from the `index.d.ts` and will cause clashes if they have corresponding `models` in the Prisma schema.

- PrismaPromise
- PrismaVersion
- JsonObject
- JsonValue
- InputJsonObject
- InputJsonValue
- PromiseType
- PromiseReturnType
- Enumerable
- RequiredKeys
- TruthyKeys
- TrueKeys
- Subset
- SelectSubset
- SubsetIntersection
- UnEnumerate
- Union 
- IntersectOf
- Overwrite
- At
- ComputeRaw
- OptionalFlat
- Strict
- Merge
- Boolean 
- True 
- False 
- Not
- Extends
- Has
- Or
- Keys
- GetScalarType
- ModelName 
- Datasources
- RejectOnNotFound 
- RejectPerModel
- RejectPerOperation 
- HasReject
- ErrorFormat 
- Hooks
- LogLevel 
- LogDefinition
- GetLogType
- GetEvents
- QueryEvent
- LogEvent
- PrismaAction 
- MiddlewareParams
- Middleware
- Datasource
- SortOrder 
- QueryMode 
- IntFilter
- StringFilter
- IntWithAggregatesFilter
- StringWithAggregatesFilter
- StringFieldUpdateOperationsInput
- IntFieldUpdateOperationsInput
- NestedIntFilter
- NestedStringFilter
- NestedIntWithAggregatesFilter
- NestedFloatFilter
- NestedStringWithAggregatesFilter
- BatchPayload

### How to reproduce

1. Generate a schema with one of the model names listed above
2. Write Prisma query
3. Observe return type is incorrect

### Prisma information

n/a

### Environment & setup

- OS: macOS
- Node.js version: 17.7.1

### Prisma Version

```
3.11
```

**Comment by psikoi** [comment]
`Metric` is now also causing conflicts with model names.

**Comment by priithaamer** [comment]
Just got hit with this by migrating prisma from `3.x` to `4.3` on a project that had a model named `Metric`.

Renaming the model worked for me but is there another workaround besides that?

**Comment by psikoi** [comment]
> Just got hit with this by migrating prisma from `3.x` to `4.3` on a project that had a model named `Metric`.
> 
> Renaming the model worked for me but is there another workaround besides that?

If I remember correctly, I had to disable `skipLibCheck` on tsconfig, but that's not really a great solution :/

**Comment by priithaamer** [comment]
> If I remember correctly, I had to disable `skipLibCheck` on tsconfig, but that's not really a great solution :/

Yes, that works as well. But i prefer model rename over this in order not to compromise type safety.

**Comment by SevInf** [comment]
All of those conflicts are fixed and all mentioned names should be useable as model name starting with upcoming Prisma 5.1 release.
